synapse/config/room_directory.py

```diff
@@ -47,7 +51,7 @@ def read_config(self, config, **kwargs):
                 _RoomDirectoryRule("room_list_publication_rules", {"action": "allow"})
             ]
 
-    def generate_config_section(self, config_dir_path, server_name, **kwargs):
+    def generate_config_section(self, config_dir_path, server_name, **kwargs) -> str:
```

Comment:
```suggestion
    def generate_config_section(self, config_dir_path: str, server_name: str, **kwargs) -> str:
```
not sure about the `kwargs` but is that right?